content/asset-tracking/index.textile

```diff
@@ -80,32 +81,36 @@ Resolution governs how often to sample locations, at what level of positional ac
+
 h2(#resolution-set). Resolution set
 
-A resolution set, is a set of resolutions that must be provided by the app in order to inform the resolution policy. While developers can define their own resolution sets, Ably also provides a default resolution set that can be used for common use cases.
+A resolution set, is a set of resolutions that must be provided by the app in order to inform the resolution policy. While developers can define their own resolution sets, Ably also provides a default resolution set that can be used with the default resolution policy.
 
 h3(#default-resolution-set). Default resolution set
 
 The @DefaultResolutionSet@ is the set of resolutions which must be defined in order to satisfy the "default resolution constraints":#default-resolution-constraints. These are required in order to inform the "default resolution policy":#default-resolution-policy.
 
 For the @DefaultResolutionSet@ the following resolutions are expected to be included:
 
-- @farWithoutSubscriber@ := The resolution to select if above the "@proximityThreshold@":#default-resolution-constraints), with no subscribers.
-- @farWithSubscriber@ := The resolution to select if above the "@proximityThreshold@":#default-resolution-constraints), with one or more subscribers.
-- @nearWithoutSubscriber@ := The resolution to select if below the "@proximityThreshold@":#default-resolution-constraints), with no subscribers.
-- @nearWithSubscriber@ := The resolution to select if below the "@proximityThreshold@":#default-resolution-constraints), with one or more subscribers.
+- @farWithoutSubscriber@ := The resolution to select if above the "@proximityThreshold@":#default-resolution-constraints, with no subscribers.
+- @farWithSubscriber@ := The resolution to select if above the "@proximityThreshold@":#default-resolution-constraints, with one or more subscribers.
+- @nearWithoutSubscriber@ := The resolution to select if below the "@proximityThreshold@":#default-resolution-constraints, with no subscribers.
+- @nearWithSubscriber@ := The resolution to select if below the "@proximityThreshold@":#default-resolution-constraints, with one or more subscribers.
 
 h2(#resolution-constraints). Resolution constraints
 
-Resolution constraints specify the thresholds and corresponding logical mappings for tracking @Resolution@s that are required by the default @ResolutionPolicy@, which can be adopted by @Publisher@ instances. While developers can define their own resolution constraints, Ably also provides default resolution constraints that can be used for common use cases.
+Resolution constraints specify the thresholds and corresponding logical mappings for tracking @Resolution@ objects that are required by the default @ResolutionPolicy@, which can be adopted by @Publisher@ instances. 
```

Comment:
To be very precise the `ResolutionConstraints` are always optional and they're not required in the default `ResolutionPolicy` implementation :wink:

Reply:
Yes, I worded that very confusingly, sorry. Does the following make more sense: "Resolution constraints specify the thresholds and corresponding logical mappings for tracking @Resolution@ objects. These @Resolution@ objects are required by the default @ResolutionPolicy@, which may be adopted by @Publisher@ instances."

Reply:
As I read it now I'm having more doubts if what's written here is true :thinking: And I've noticed that we have the same description written in the kdocs in our code :thinking: 

**This is how I understand this sentence** (please correct me if I'm wrong):
We have a thing called "resolution constraints". It specifies some thresholds and logical mappings. Those thresholds and mappings are then used for tracking some "resolution objects". The "resolution objects" are required by the default "Resolution Policy".

**A detailed description of how it works in the code**:
We have a thing called "resolution constraints". It specifies:
-  the proximity threshold
-  a set of resolutions that are used depending on some factors (i.e. threshold)
- the battery level threshold together with a low battery multiplier (applied for desired interval if battery is below the threshold)

This "resolution constraints" is an optional part of a `Trackable` object. The `Trackable` object represents a thing that the `Publisher` is tracking. If we create a `Trackable` object with those "resolution constraints" then they will be used by `ResolutionPolicy` in order to resolve the best possible `Resolution` for that `Trackable`. However, if the "resolution constraints" are not provided, the `ResolutionPolicy` still works and resolves a valid `Resolution` but its calculations are simpler.

**How it's described in our code docs:**

``` kotlin
/**
 * Specifies the thresholds and corresponding logical mappings for tracking [Resolution]s that are required by the
 * default [ResolutionPolicy], which can adopted by [Publisher] instances using the [Builder][Publisher.Builder]'s
 * [resolutionPolicy][Publisher.Builder.resolutionPolicy] method, providing it with an instance of the
 * [DefaultResolutionPolicyFactory] class.
 */
```

@QuintinWillison given my above description of how it works in the code, are our docs accurate or should we change them? :thinking:

Reply:
I actually think the wording is wrong now. 

`ResolutionConstraints` are defined on per `Trackable` basis, and the are used to calculate appropriate `Resolution` by the `ResolutionPolicy`.

Reply:
@KacperKluka do you think what I wrote above makes sense?

Reply:
Yes, it's how it works in the code and how I understand it too :wink:

Reply:
@tbedford we probably should change the wording to express what I wrote 2 comments above.

Reply:
@kavalerov I think I did already! Can you double check please.

Reply:
FYI I've created an [issue](https://github.com/ably/ably-asset-tracking-android/issues/305) in the android repo to fix code docs :wink: